CMakeLists.txt

```diff
@@ -18,6 +18,14 @@ if(PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR)
   )
 endif()
 
+# ---- Project settings ----
+
+if(NOT DEFINED CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS NO)
+endif()
+
```

Comment:
I prefer setting options directly on targets vs changing global project settings. Global settings would also leak to any included subdirectories and dependencies, which may not always be desirable.

Reply:
IMHO today, this is a really good starting default setting

and be explicit, better than using the compiler defaults

Reply:
These global options should however be defined by the end-user, not libraries. Also they can simply be passed to CMake during configuration.

Reply:
you mean it is OK for an end-user that you build the lib with gnu exteions, and **`-std=g++20`**, because you build it wiht a modern gcc?

What happens, wenn the lib is used by an end-user via `brew `or `conan`?

Reply:
> you mean it is OK for an end-user that you build the lib with gnu exteions, and -std=g++20, because you build it wiht a modern gcc?

If that are his default compiler settings and he doesn't have the need to override them, I don't see why not?

> What happens, wenn the lib is used by an end-user via brew or conan?

When a library get's added to brew / Conan / vcpkg it uses an external install script that forwards any configuration flags as required by the package manager to CMake. It's not a library issue.

Reply:
OK